### PR TITLE
Add nucliadb tests for paragraph page position and duplicates

### PR DESCRIPTION
**Pull request by lferran**
### Description
- Adds nucliadb e2e tests for:
  - paragraph deduplication on search results
  - paragraph positions on search results

- Fixes computing duplicates paragraphs

### CODE REVIEW

File: nucliadb/nucliadb/ingest/tests/orm/test_brain.py [hunk 1]
```diff
@@ -32,46 +33,83 @@
...
         else:
             assert paragraph.repeated_in_field is True
 
 
 def test_duplicate_paragraph_checker():
-    p1 = Paragraph(text="repeated_text")
-    p2 = Paragraph(text="some_text")
-    p3 = Paragraph(text="repeated_text")
-
     dupcheck = DuplicateParagraphsChecker()
```
Comment (jotare) [inline nucliadb/nucliadb/ingest/tests/orm/test_brain.py:106]: I think `DuplicateParagraphsChecker` is an unneeded abstraction. Python built in `set` already hashes its keys to be able to lookup elements in O(1).

File: nucliadb/nucliadb/ingest/orm/resource.py [hunk 2]
```diff
@@ -260,8 +260,14 @@ async def generate_index_message(self) -> ResourceBrain:
                 if type_id == FieldType.FILE and isinstance(field, File):
                     page_positions = await get_file_page_positions(field)
 
+                extracted_text = await field.get_extracted_text()
                 brain.apply_field_metadata(
-                    field_key, field_metadata, [], {}, page_positions=page_positions
+                    field_key,
+                    field_metadata,
+                    [],
+                    {},
```
Comment (jotare) [inline nucliadb/nucliadb/ingest/orm/resource.py:268]: NIT: add here parameter names to read easily what are this empty list and dict

**Review by jotare — APPROVED** [review]

File: nucliadb/nucliadb/ingest/tests/orm/test_brain.py (hunk 1 above)
Comment (lferran) [reply nucliadb/nucliadb/ingest/tests/orm/test_brain.py:106]: Agree! It is totally useless. I will remove it
